docs/typescript.md

```diff
@@ -1,29 +1,76 @@
+
+NOTE: Re-run the `dnscontrol write-types` command any time you upgrade
+dnscontrol. Because it is generated from the command, it will always be correct
+for the version of DNSControl you are using.
+
+2. Tell your editor
+
+At this point some features (autocomplete) will work. However to get the full experience, including
+type checking (i.e. red squiggly underlines when you misuse APIs), there is one more step.
+
+Add this comment to the top of your `dnsconfig.js` file:
+
+```
+// @ts-check
+```
 
-That should be all you need to do! If you’re using VS Code (or another editor that supports TypeScript), you should now be able to see the type information in your `dnsconfig.js` file as you type. Hover over record names to read their documentation without having to open the website!
+That should be all you need to do!
+
+If your edit requires extra steps, please [file a bug](https://github.com/StackExchange/dnscontrol/issues) and we'll update this page.
+
+### Bugs?
+
+**Bugs?**  Not all features of DNSControl work perfectly at the moment. Please report bugs and feature requests on https://github.com/StackExchange/dnscontrol/issues
+
+**This is experimental.** This feature is currently experimental. We might change the installation instructions as we find better ways to enable this.
+
+## Known bugs
+
+## Bug: `CLI_DEFAULTS` not implemented
+
+Bug: Values passed to `CLI_DEFAULTS` (and the corresponding `-v` command-line option) don’t show up as global variables
+
+Workaround: create a new `.d.ts` file in the same folder as your `dnsconfig.js` file. In that file, add the following line for each variable you want to use (replacing `VARIABLE_NAME` with the name of the variable. 
+
+```
+declare const VARIABLE_NAME: string;
+```
 
-## Type Checking
+This will tell TypeScript that the variable exists, and that it’s a string.
 
-If you add the comment `// @ts-check` to the top of your `dnsconfig.js` file, you can enable _type checking_ for your DNSControl configuration. This will allow your editor’s integrated version of TypeScript to check your configuration for possible mistakes in addition to providing enhanced autocomplete. Note that not all features of DNSControl work perfectly at the moment.
+## Bug: `FETCH` not always accurate
 
-Specifically:
+Bug: `FETCH` is always shown as available, even if you don’t run DNSControl with the `--allow-fetch` flag.
 
--   Values passed to `CLI_DEFAULTS` (and the corresponding `-v` command-line option) don’t show up as global variables
-    -   Workaround: create a new `.d.ts` file in the same folder as your `dnsconfig.js` file. In that file, add the following line: <code>declare const _variableName_: string;</code> for each variable you want to use. This will tell TypeScript that the variable exists, and that it’s a string.
--   `FETCH` is always shown as available, even if you don’t run DNSControl with the `--allow-fetch` flag.
+Workaround: None
```

Comment:
Based on https://github.com/StackExchange/dnscontrol/pull/1940/files#r1070092563, delete this line.

```suggestion
```